File: jetpack-udemy/Dependency/Dagger2Singleton/app/src/main/java/com/androidtutz/anushka/didemo/MainActivity.java
package com.androidtutz.anushka.didemo;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {
    @Inject
    SmartPhone smartPhone;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        App.getApp().getSmartPhoneComponent().inject(this);

        smartPhone.makeACall();

        Button button=findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               Intent intent=new Intent(MainActivity.this,Main2Activity.class);
               startActivity(intent);
            }
        });

    }
}

File: jetpack-udemy/Dependency/StateOfModule/app/src/main/java/com/anushka/didemo/MainActivity.kt
package com.anushka.didemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import javax.inject.Inject

class MainActivity : AppCompatActivity() {
@Inject
lateinit var smartPhone: SmartPhone

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        DaggerSmartPhoneComponent.create()
//            .inject(this)
//        smartPhone.makeACallWithRecording()
          DaggerSmartPhoneComponent.builder()
              .memoryCardModule(MemoryCardModule(1000))
              .build()
              .inject(this)
    }
}
